src/dom-utils.ts

```diff
@@ -27,55 +27,77 @@ export function moveFocusToDialog(el: HTMLElement) {
   focused.focus()
 }
 
-/**
- * Get focusable children by recursively traversing the subtree of the given
- * element, while accounting for Shadow DOM subtrees
- */
-export function getFocusableChildren(el: ParentNode): HTMLElement[] {
-  // Check for the base case of our recursion
-  if (el instanceof HTMLElement) {
-    // If this element is marked as inert, neither it nor any member of its
-    // subtree will be focusable, therefore there are no focusable children
-    if (el.inert) return []
+// Get the first and last focusable elements in a given subtree
+export function getFirstAndLastFocusableChild(el: HTMLElement) {
+  const first = findFocusableElement(el, true)
+  let last = null
+  if (first !== null) {
+    last = findFocusableElement(el, false) || first
+  }
+  return [first, last]
+}
```

Comment:
```suggestion
export function getFirstAndLastFocusableChild(el: HTMLElement) {
  const first = findFocusableElement(el, true)
  const last = findFocusableElement(el, false)
  
  return [first, last || first]
}
```

Or did I miss something?

Reply:
The way I have it now, we're saving ourselves from extra walking if there is only one or 0 focusable elements in the given DOM. No need to start a backwards walk if the forward walk does not find anything at alll.